private/bufpkg/bufimage/bufimagemodify/field_option.go

```diff
@@ -132,3 +124,27 @@ func isJsTypePermittedForType(fieldType descriptorpb.FieldDescriptorProto_Type)
 		fieldType == descriptorpb.FieldDescriptorProto_TYPE_FIXED64 ||
 		fieldType == descriptorpb.FieldDescriptorProto_TYPE_SFIXED64
 }
+
+func isFieldOptionDisabledForFile(
```

Comment:
I think it's a bit misleading to name this `isFieldOptionDisabledForFile`, for two reasons:

- you are not checking a single field option, you are checking the all disable rules in the config
- you are also not only checking it for the file, you are also checking the specified field

I would probably put the checks in a helper and then use them in the loops above, something like:

```go

func isFieldOptionDisabledForFile(
  disableRule bufconfig.ManagedDisableRule,
  imageFile bufimage.ImageFile,
  fieldOption bufconfig.FieldOption,
)  bool {
  if disableRule.FileOption() != bufconfig.FileOptionUnspecified {
    return false // FileOption specified, not a matching rule.
	}
  if disableRule.FieldOption() != bufconfig.FieldOptionUnspecified && disableRule.FieldOption() != fieldOption {
	return false // FieldOption specified, does not match option.
  }
  return !fileMatchConfig(imageFile, disableRule.Path(), disableRule.ModuleFullName())
}

func isFieldOptionDisabledForField(
  disableRule bufconfig.ManagedDisableRule,
  imageFile bufimage.ImageFile,
  fieldOption bufconfig.FieldOption,
  fullName protoreflect.FullName,
) bool {
  // Check if the field option is disabled for the full file first, if so, we can return here.
  if isFieldOptionDisabledForFile(disableRule, imageFile, fieldOption) {
    return true
  }
  return disableRule.FieldName() != "" && disableRule.FieldName() != string(fullName)
}
```

And then I would restore the `for` loops above to look something like:

```go
for _, disableRule := range config.Disables() {
  if isFieldOptionDisabledForFile(disableRule, imageFile, fieldOption) {
    return nil
  }
}

for _, disable := range config.Disables() {
  if isFieldOptionDisabledForField(disableRule, imageFile, fieldOption, fullName) {
    return nil
  }
}
```

Reply:
Fixed. The `ForField` one must have the FieldName exclusion check before the other cases.